kokoro/gcp_ubuntu/cmake/build.sh

```diff
@@ -22,17 +22,12 @@ set -x
 # Print the UTC time when set -x is on
 export PS4='[$(date -u "+%T %Z")] '
 
-export CMAKE_BIN="$(which cmake)"
-export CC="$(which clang-6.0)"
-export CXX="$(which clang++-6.0)"
-
 # Check these exist and print the versions for later debugging
-"$CMAKE_BIN" --version
-"$CC" --version
-"$CXX" --version
+cmake --version
+clang --version
+clang++ --version
+python3 --version
```

Comment:
The cmake build uses the environment variables. `CC` and `CXX` are standard env variables. It might default to the system version if they're not specified, but it's still good to be explicit. `CMAKE_BIN` is our invention (https://sourcegraph.com/github.com/google/iree/-/blob/build_tools/cmake/build_docs.sh#L26). Not sure whether the cmake build uses `PYTHON_BIN`. It might be our own special invention.

Anyway I think it's good to use the explicit env variables for the things where they actually control the version.

Reply:
That makes sense. I didn't see anything about it in the getting started doc so I thought that was specific to Kokoro. I'll add those back in.

Is it okay to keep the specific configuration flags (`DCMAKE_C/CXX_COMPILER=clang/++`) in `rebuild.sh`? Since these flags aren't mentioned in our cmake docs the errors that their absence cause when trying to use FastBuild are pretty confusing.

Reply:
Ah I missed those. I think we should mention setting the env variables in the getting started docs instead. I think we want the build scripts to be reasonably reusable so that someone can (try to) do the same thing, but with GCC. That's why CMAKE_BIN is an env variable there. I was testing with various cmake versions and it was super annoying that that wasn't injectable. But I could be persuaded that putting everything in the cmake command is the way to go instead.

Reply:
@ScottTodd thoughts?

Reply:
I'm not really following the conversation and what all the scripts are doing, but the [getting started docs](https://google.github.io/iree/GetStarted/LinuxCMake#build) do mention setting `-DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++`. Are there other env vars we should mention?

Reply:
A previous incarnation of the docs instead said to set `CC` and `CXX`, which applies to both cmake and bazel I think. My suggestion is that we use these env variables in the build scripts instead of writing it into the cmake invocation because it's something that's in the user's control and allows things like trying other clang versions while still invoking through the script. We can have the build script default them to `$(which clang)` and `$(which clang++)` like it defaults `CMAKE_BIN` to `$(which cmake)`. Potentially we should also recommend the env variables instead of the cmake flags in the getting started docs?

Reply:
The [Bazel docs](https://google.github.io/iree/GetStarted/LinuxBazel#install-a-compiler) suggest setting `CC` and `CXX`. I think CMake flags are slightly easier to configure in IDEs than env vars.

Reply:
I'm fine with whatever makes sense for the getting started docs. For these build scripts I think the best scenario is that you can control them with env variables but they still work without (sorry Phoenix I didn't realize the instructions had changed to no longer mention those variables). So we can just do the same thing we do with CMAKE_BIN and default it to something reasonable but use the env variables if specified. Anyway I can do that in a followup